src/wei/types/step_types.py

```diff
@@ -29,48 +29,46 @@ class StepResponse(BaseModel):
     in response to action requests
     """
 
-    action_response: StepStatus = StepStatus.SUCCEEDED
-    """Whether the action succeeded, failed, is running, or is idle"""
-    action_msg: str = ""
-    """Any result from the action. If the result is a file, this should be the file name"""
-    action_log: str = ""
-    """Error or log messages resulting from the action"""
+    status: StepStatus = StepStatus.SUCCEEDED
+    """Whether the step succeeded or failed"""
+    error: Optional[str] = None
+    """Error message resulting from the action"""
+    data: Optional[dict] = None
+    """Key value dict of data returned from step"""
+    files: Optional[dict] = None
+    """Key value dict of file labels and file names from step"""
 
     def to_headers(self) -> Dict[str, str]:
         """Converts the response to a dictionary of headers"""
         return {
-            "x-wei-action_response": str(self.action_response),
-            "x-wei-action_msg": self.action_msg,
-            "x-wei-action_log": self.action_log,
+            "x-wei-status": str(self.status),
+            "x-wei-data": json.dumps(self.data),
+            "x-wei-error": json.dumps(self.error),
+            "x-wei-files": json.dumps(self.files),
         }
 
     @classmethod
     def from_headers(cls, headers: Dict[str, Any]) -> "StepResponse":
         """Creates a StepResponse from the headers of a file response"""
 
         return cls(
-            action_response=StepStatus(headers["x-wei-action_response"]),
-            action_msg=headers["x-wei-action_msg"],
-            action_log=headers["x-wei-action_log"],
+            status=StepStatus(headers["x-wei-status"]),
+            error=json.loads(headers["x-wei-error"]),
+            files=json.loads(headers["x-wei-files"]),
+            data=json.loads(headers["x-wei-data"]),
         )
 
     @classmethod
-    def step_succeeded(cls, action_msg: str, action_log: str = "") -> "StepResponse":
+    def step_succeeded(
+        cls, files: Dict[str, str] = None, data: Dict[str, str] = None
+    ) -> "StepResponse":
         """Returns a StepResponse for a successful step"""
-        return cls(
-            action_response=StepStatus.SUCCEEDED,
-            action_msg=action_msg,
-            action_log=action_log,
-        )
+        return cls(status=StepStatus.SUCCEEDED, files=files, data=data)
 
     @classmethod
-    def step_failed(cls, action_log: str, action_msg: str = "") -> "StepResponse":
+    def step_failed(cls, error: str) -> "StepResponse":
         """Returns a StepResponse for a failed step"""
-        return cls(
-            action_response=StepStatus.FAILED,
-            action_msg=action_msg,
-            action_log=action_log,
-        )
+        return cls(error=error)
 
 
 class StepFileResponse(FileResponse):
```

Comment:
This should probably take `files` as a dict of PathLikes and then, if there's more than one, automatically zip them and return them using https://docs.python.org/3/library/tempfile.html